shared/src/commonMain/kotlin/org/hyperskill/app/topics_to_discover_next/domain/interactor/TopicsToDiscoverNextInteractor.kt

```diff
@@ -38,12 +40,19 @@ class TopicsToDiscoverNextInteractor(
                     .getCurrentProfile(primarySourceType = DataSourceType.CACHE)
                     .getOrThrow()
 
-                if (currentProfile.isCurrentTrackCompleted) {
+                if (currentProfile.isCurrentTrackCompleted || currentProfile.trackId == null) {
                     return@runCatching emptyList()
                 }
 
+                val studyPlan = trackInteractor
+                    .getStudyPlanByTrackId(currentProfile.trackId)
```

Comment:
Instead of loading all study plans (see `TrackRemoteDataSourceImpl.getStudyPlans() -> /api/study-plans`) we can load only the current one `/api/study-plans/current`. This is precisely what we are looking for and it works faster.

Reply:
Also, I think we can create `study_plan` data module and move this logic from `track` module, because we are planning to work on study plan feature soon.